tooling/api/src/main/java/org/jboss/windup/tooling/ExecutionBuilderSetOptionsAndProgressMonitor.java

```diff
@@ -0,0 +1,36 @@
+package org.jboss.windup.tooling;
+
+import org.jboss.windup.exec.WindupProgressMonitor;
+
+/**
+ * Allows setting windup options, including the {@link WindupProgressMonitor}.
+ *
+ * @author <a href="mailto:jesse.sightler@gmail.com">Jesse Sightler</a>
+ */
+public interface ExecutionBuilderSetOptionsAndProgressMonitor
+{
+    /**
+     * Sets a pattern of file paths to ignore during processing.
+     */
+    ExecutionBuilderSetOptions ignore(String ignorePattern);
+
+    /**
+     * Sets the package name prefixes to scan (the default is to scan all packages).
+     */
+    ExecutionBuilderSetOptions includePackage(String includePackagePrefix);
```

Comment:
Should these be genericised such that the UserInputOption is used instead? This seems very specific to the application migration stuff.